src/App.js

```diff
@@ -1,16 +1,50 @@
 import React from 'react';
 import './App.css';
 import chatMessages from './data/messages.json';
+import ChatLog from './components/ChatLog';
+import { useState } from 'react';
 
 const App = () => {
+  const [entries, setEntries] = useState(chatMessages);
+
+  const updateLiked = (id) => {
+    const newEntries = [];
+    for (const entry of entries) {
+      const newEntry = { ...entry };
+      if (newEntry.id === id) {
+        newEntry.liked = !newEntry.liked;
+      }
+      newEntries.push(newEntry);
+    }
+    setEntries(newEntries);
+  };
+
+  const countLikes = () => {
+    let totalHearts = 0;
+    for (const entry of entries) {
+      if (entry.liked === true) {
+        totalHearts += 1;
+      }
+    }
+    return totalHearts;
+  };
+
   return (
     <div id="App">
       <header>
-        <h1>Application title</h1>
+        <h1>
+          Chat between {chatMessages[0].sender} and {chatMessages[1].sender}
+        </h1>
+        <section>
+          <h2 className="widget" id="heartWidget">
+            {countLikes()} ❤️s
+          </h2>
+        </section>
```

Comment:
Ah, this is a little awkward (and the example page might be wrong) but this shouldn't be a `h2` since it isn't part of the logical "table of contents" of the page.

```suggestion
        <p className="widget" id="heartWidget">
          {countLikes()} ❤️s
        <p>
```

Using something like a `p` or a `span` or a `div` is probably more appropriate here.